src/Views/Form/Themes/Sequoia/assets/js/form.js

```diff
@@ -1,53 +1,197 @@
+				if ( index !== step ) {
+					return obj.selector;
+				}
+			} );
+			const hideSelector = hide.filter( Boolean ).join( ', ' );
 
-			// Show remain form options.
-			$( 'form > *:not(.give-section.choose-amount)', $container ).show();
+			$( hideSelector ).hide();
+			$( steps[ step ].selector ).show();
 
-			$parent.removeClass( 'give-showing_choose-amount-section' ).addClass( 'give-showing__personal-section' );
-		}
+			if ( step === steps.length - 1 ) {
+				$advanceButton.hide();
+			} else {
+				$advanceButton.show();
+			}
+
+			steps[ step ].setup();
+			navigator.currentStep = step;
+		},
+		back: () => {
+			const prevStep = navigator.currentStep !== 0 ? navigator.currentStep - 1 : 0;
+			navigator.goToStep( prevStep );
+			navigator.currentStep = prevStep;
+		},
+		forward: () => {
+			const nextStep = navigator.currentStep !== null ? navigator.currentStep + 1 : 1;
+			navigator.goToStep( nextStep );
+			navigator.currentStep = nextStep;
+		},
+	};
+
+	const steps = [
+		{
+			id: 'introduction',
+			title: null,
+			selector: '.give-section.introduction, .give-section.income-stats, .give-section.progress-bar',
+			label: templateOptions.introduction.donate_label,
+			showErrors: false,
+			setup: () => {
+
+			},
+		},
+		{
+			id: 'choose-amount',
+			title: templateOptions.payment_amount.header_label,
+			selector: '.give-section.choose-amount',
+			label: templateOptions.payment_amount.next_label,
+			showErrors: false,
+			setup: () => {
+				$( '.give-donation-level-btn' ).each( function() {
+					const hasTooltip = $( this ).attr( 'has-tooltip' );
+					if ( hasTooltip ) {
+						return;
+					}
+
+					const value = $( this ).attr( 'value' );
+					const text = $( this ).text();
+					if ( value !== 'custom' ) {
+						const wrap = `<span class="give-tooltip hint--top hint--bounce" style="width: 100%" aria-label="${ text }" rel="tooltip"></span>`;
+						const symbol = $( '.give-currency-symbol' ).text();
+						const position = $( '.give-currency-symbol' ).hasClass( 'give-currency-position-before' ) ? 'before' : 'after';
+						const html = position === 'before' ? `<div class="currency">${ symbol }</div>${ value }` : `${ value }<div class="currency">${ symbol }</div>`;
+						$( this ).html( html );
+						$( this ).wrap( wrap );
+						$( this ).attr( 'has-tooltip', true );
+					}
+				} );
+				$( '.give-total-wrap', $container ).addClass( 'give-flex' );
+				$( '.give-donation-levels-wrap', $container ).addClass( 'give-grid' );
+			},
+		},
+		{
+			id: 'payment',
+			title: templateOptions.payment_information.header_label,
+			label: templateOptions.payment_information.checkout_label,
+			selector: '.give-section.payment',
+			showErrors: true,
+			setup: () => {
+				$( '.give-label' ).html( '' );
+				$( 'label[for=give-first]' ).html( '<i class="fas fa-user"></i>' );
+				$( 'label[for=give-email]' ).html( '<i class="fas fa-envelope"></i>' );
+			},
+		},
+	];
+
+	const styles = {
```

Comment:
@henryholtgeerts what's the advantage of using CSS in JS rather than generating it in PHP?

Reply:
Good question - primary advantage is that it helps avoid utilizing too many obscure hooks in the Actions.php file of Sequoia. 

As I started work on these styles, it became clear that rather than spending time digging around in template.php for all the appropriate hooks I could simply be loading template options into the form with javascript - which saves time and helps cut down on GiveWP's chronic memory overuse.

Reply:
From my vantage point, the primary advantage is tree-shaking. When generating CSS in PHP you only have the context of the server to work with, so you generate the CSS files to the best of your ability and knowledge at that point. This is the best route at times. CSS in JS, on the other hand, allows you to couple your styles to your components, so when compiling the assets _only_ the components that are actually used (and their styles) are included in the resulting script. This means it has the chance for the sum js and css file sizes to be smaller.